optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +255,170 @@ public void skipAhead() throws ExecutionException, InterruptedException {
         assertTrue(eventCount.get() < 4);
     }
 
+    @Ignore("https://issues.redhat.com/browse/PLANNER-1837")
+    @Test
+    public void solveMultipleThreadedMovesWithSolverManager_allGetSolved() throws ExecutionException, InterruptedException {
+        int processCount = Runtime.getRuntime().availableProcessors();
+        final SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class)
+                .withPhases(new ConstructionHeuristicPhaseConfig(), new LocalSearchPhaseConfig())
+//                .withTerminationConfig(new TerminationConfig().withSecondsSpentLimit(4L))
+                // Adds moveThreadCount to the solver config.
+                .withMoveThreadCount("AUTO");
+        // Creates solverManagerConfig with multiple threads.
+        SolverManager<TestdataSolution, Integer> solverManager = SolverManager.create(solverConfig, new SolverManagerConfig());
+
+        List<SolverJob<TestdataSolution, Integer>> jobs = new ArrayList<>();
+        for (int i = 0; i < processCount; i++) {
+            jobs.add(solverManager.solve(i, PlannerTestUtils.generateTestdataSolution("s" + i, 10)));
+        }
+
+        assertInitializedJobs(jobs);
+    }
+
+    private void assertInitializedJobs(List<SolverJob<TestdataSolution, Integer>> jobs) throws InterruptedException, ExecutionException {
+        for (SolverJob<TestdataSolution, Integer> job : jobs) {
+            // Method getFinalBestSolution() waits for the solving to finish, therefore it ensures synchronization.
+            assertSolutionInitialized(job.getFinalBestSolution());
+        }
+    }
+
+    @Test(timeout = 2_000)
+    public void submitMoreProblemsThanCpus_allGetSolved() throws InterruptedException, ExecutionException {
+        // Use twice the amount of problems than available processors.
+        int problemCount = Runtime.getRuntime().availableProcessors() * 2;
+
+        SolverManager<TestdataSolution, Integer> solverManager = createSolverManagerTestableByDifferentConsumers();
+        assertDifferentSolveMethods(problemCount, solverManager);
+    }
+
+    private SolverManager<TestdataSolution, Integer> createSolverManagerTestableByDifferentConsumers() {
+        final SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class)
+                .withPhases(createGraduallyInitializedPhaseConfigs());
+
+        SolverManagerConfig solverManagerConfig = new SolverManagerConfig();
+
+        return SolverManager.create(solverConfig, solverManagerConfig);
+    }
+
+    private PhaseConfig[] createGraduallyInitializedPhaseConfigs() {
+        PhaseConfig[] configs = new PhaseConfig[2];
+        for (int x = 0; x < 2; x++) {
+            configs[x] = createSetXEntityToXValuePhaseConfig(x);
+        }
+        return configs;
+    }
+
+    private CustomPhaseConfig createSetXEntityToXValuePhaseConfig(int x) {
```

Comment:
I wouldn't extract this method (see other case below). No DRY and no readability improvement.  createSetXEntityToXValuePhaseConfig isn't an obvious or near-obvious abstraction